packages/discovery/src/discovery/interactive/InteractiveDiffIgnore.ts

```diff
@@ -0,0 +1,196 @@
+import select from '@inquirer/select'
+import { ContractParameters } from '@l2beat/discovery-types'
+import chalk from 'chalk'
+
+import { DiscoveryOverridesBuilder } from '../config/DiscoveryOverridesBuilder'
+import { DiscoveryDiff } from '../output/diffDiscovery'
+
+interface Change {
+  contract: string
+  property: string
+}
+
+export class InteractiveDiffIgnore {
+  constructor(private readonly dob: DiscoveryOverridesBuilder) {}
+
+  async runForDiffs(discoveryDiffs: DiscoveryDiff[]): Promise<void> {
+    console.log(chalk.green('Starting interactive mode...'))
+
+    // Filter-out diffs without changes
+    const changesWithDiffs: Change[] = discoveryDiffs.flatMap(
+      (discoveryDiff) => {
+        const contract = discoveryDiff.name
+          ? discoveryDiff.name
+          : discoveryDiff.address.toString()
+
+        if (!discoveryDiff.diff) {
+          return []
+        }
+
+        return discoveryDiff.diff.map((d) => ({
+          contract,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          property: d.key!,
+        }))
+      },
+    )
+
+    const rootPropsChanged = getRootPropertiesChanged(changesWithDiffs)
+
+    for (const { property, contract: addressOrName } of rootPropsChanged) {
+      const contract = this.getContract(addressOrName)
+
+      if (!contract) {
+        const msg = `Skipping ${chalk.italic.blue(
+          property,
+        )} - contract not found`
+
+        console.log(msg)
+        continue
+      }
+
+      const prettyContractName = getNameWithColor(contract)
+
+      const isIgnoredInWatchMode = this.dob
+        .getWatchMode(contract)
+        .includes(property)
+
+      if (isIgnoredInWatchMode) {
+        const msg = `Skipping ${chalk.italic.blue(
+          property,
+        )} on ${prettyContractName} - ignored in watch mode`
+        console.log(msg)
+        continue
+      }
+
+      const isIgnoredMethod = this.dob
+        .getIgnoredMethods(contract)
+        .includes(property)
+
+      if (isIgnoredMethod) {
+        const msg = `Skipping ${chalk.italic.blue(
+          property,
+        )} on ${prettyContractName}- ignored method`
+        console.log(msg)
+        continue
+      }
+
+      const isCustomHandler = this.dob.isCustomHandler(contract, property)
+
+      if (isCustomHandler) {
+        const msg = `Skipping ${chalk.italic.blue(
+          property,
+        )} on ${prettyContractName} - custom handler`
+        console.log(msg)
+        continue
+      }
+
+      await this.configureSingleProperty(contract, property)
+    }
+
+    console.log(chalk.green('Flushing overrides...'))
+    await this.dob.flushOverrides()
```

Comment:
Couple of things

- we should save inside the UI
- `dob` shouldn't call `writeFileSync`, change the name from `flushOverrides` to `buildConfig` which returns a new config with overrides that you can later save manually inside the `runDiscovery`